builtin/v10/migration/miner.go

```diff
@@ -0,0 +1,454 @@
+				PartitionsPoSted:                  inDeadline.PartitionsPoSted,
+				EarlyTerminations:                 inDeadline.EarlyTerminations,
+				LiveSectors:                       inDeadline.LiveSectors,
+				TotalSectors:                      inDeadline.TotalSectors,
+				FaultyPower:                       miner10.PowerPair(inDeadline.FaultyPower),
+				OptimisticPoStSubmissions:         inDeadline.OptimisticPoStSubmissions,
+				SectorsSnapshot:                   outSectorsSnapshotCid,
+				PartitionsSnapshot:                inDeadline.PartitionsSnapshot,
+				OptimisticPoStSubmissionsSnapshot: inDeadline.OptimisticPoStSubmissionsSnapshot,
+			}
+
+			outDlCid, err := store.Put(ctx, &outDeadline)
+			if err != nil {
+				return cid.Undef, xerrors.Errorf("saving deadline to store: %w", err)
+			}
+
+			outDeadlines.Due[i] = outDlCid
+		}
+	}
+
+	return store.Put(ctx, &outDeadlines)
+}
+
+func (m minerMigrator) migrateParitions(ctx context.Context, store adt9.Store, cache MigrationCache, partitions cid.Cid) (cid.Cid, error) {
+	return cache.Load(PartitionsAmtKey(partitions), func() (cid.Cid, error) {
+		inPartitionsArr, err := adt9.AsArray(store, partitions, miner9.DeadlinePartitionsAmtBitwidth)
+		if err != nil {
+			return cid.Undef, xerrors.Errorf("failed to load partitions: %w", err)
+		}
+
+		outPartitions, err := adt10.MakeEmptyArray(store, miner10.DeadlinePartitionsAmtBitwidth)
+		if err != nil {
+			return cid.Undef, xerrors.Errorf("failed to create new partitions: %w", err)
+		}
+
+		var inPartition miner9.Partition
+		err = inPartitionsArr.ForEach(&inPartition, func(i int64) error {
+			outEsCid, err := m.migrateExpirations(ctx, store, cache, inPartition)
+			if err != nil {
+				return xerrors.Errorf("failed to migrate expirations: %w", err)
+			}
+
+			outPartition := miner10.Partition{
+				Sectors:           inPartition.Sectors,
+				Unproven:          inPartition.Unproven,
+				Faults:            inPartition.Faults,
+				Recoveries:        inPartition.Recoveries,
+				Terminated:        inPartition.Terminated,
+				ExpirationsEpochs: outEsCid,
+				EarlyTerminated:   inPartition.EarlyTerminated,
+				LivePower:         miner10.PowerPair(inPartition.LivePower),
+				UnprovenPower:     miner10.PowerPair(inPartition.UnprovenPower),
+				FaultyPower:       miner10.PowerPair(inPartition.FaultyPower),
+				RecoveringPower:   miner10.PowerPair(inPartition.RecoveringPower),
+			}
+
+			err = outPartitions.Set(uint64(i), &outPartition)
+			if err != nil {
+				return xerrors.Errorf("failed to set partition: %w", err)
+			}
+
+			return nil
+		})
+		if err != nil {
+			return cid.Undef, xerrors.Errorf("failed to iterate over partitions: %w", err)
+		}
+
+		return outPartitions.Root()
+	})
+}
+
+func (m minerMigrator) migrateExpirations(ctx context.Context, store adt9.Store, cache MigrationCache, inPartition miner9.Partition) (cid.Cid, error) {
+	return cache.Load(ExpirationsAmtKey(inPartition.ExpirationsEpochs), func() (cid.Cid, error) {
```

Comment:
Can we think through whether it's worth caching  Expirations? I suspect it's true that in almost every case if the Partition cache has a cache miss, so will the Expirations cache (and the same is true of hits).

But I don't know that for a fact -- look into it?

Reply:
You seem to be right. Removed the caching for expiration queues.